### ISSUE TITLE
C call strftime from localtime error

### ISSUE THREAD

**Issue by iamdrq** [maiyao1988/ExAndroidNativeEmu#58]
JAVA JNI:
mynative-lib.cpp
```
#include <jni.h>
#include <string>
#include <time.h>

extern "C" JNIEXPORT jstring JNICALL
Java_com_miui_calculator_MainActivity_stringFromJNI(
        JNIEnv* env,
        jobject /* this */) {

    time_t rawtime;
    struct tm *info;
    char buffer[80];

    time( &rawtime );
    info = localtime( &rawtime );

    strftime(buffer, 80, "%Y-%m-%d %H:%M:%S", info);
    std::string hello = buffer;

    return env->NewStringUTF(hello.c_str());
}
```

ExAndroidNativeEmu:

```

emulator = Emulator(
    vfs_root=posixpath.join(posixpath.dirname(__file__), "vfs")
)

lib_module = emulator.load_library("libmynative-lib.so")

try:

    res = emulator.call_symbol(lib_module, 'Java_com_miui_calculator_MainActivity_stringFromJNI', emulator.java_vm.jni_env.address_ptr, 0x00)
    print("Response from JNI call: %s" % res)

except UcError as e:
    print("Exit at %x" % emulator.mu.reg_read(UC_ARM_REG_PC))
    raise

```
Call error:
```
2022-02-18 18:31:28,848   DEBUG                               root | 22868 syscall 6 lr=0xCBC12653
2022-02-18 18:31:28,848   DEBUG                               root | 22868 Executing syscall close(0x0000000C) at 0xCBC40854
Exit at cbc1644c
Traceback (most recent call last):
  File "E:\Android\ExAndroidNativeEmu\mytest.py", line 59, in <module>
    res = emulator.call_symbol(lib_module, 'Java_com_miui_calculator_MainActivity_stringFromJNI', emulator.java_vm.jni_env.address_ptr, 0x00)
  File "E:\Android\ExAndroidNativeEmu\androidemu\emulator.py", line 259, in call_symbol
    return self.call_native(symbol_addr, *argv)
  File "E:\Android\ExAndroidNativeEmu\androidemu\emulator.py", line 265, in __call_native32
    self.__sch.exec(addr)
  File "E:\Android\ExAndroidNativeEmu\androidemu\scheduler.py", line 283, in exec
    self.__emu.mu.emu_start(start_pos, self.__stop_pos, 0, 0)
  File "D:\Python39\lib\site-packages\unicorn\unicorn.py", line 318, in emu_start
    raise UcError(status)
unicorn.unicorn.UcError: Invalid memory read (UC_ERR_READ_UNMAPPED)
```
ENV:
I used the libc.so form this project issues( because the project run all test happen : unicorn.unicorn.UcError: Invalid memory read (UC_ERR_READ_UNMAPPED)):
https://github.com/AeonLucid/AndroidNativeEmu/blob/master/examples/example_binaries/32/libc.so

[libmynative-lib.zip](https://github.com/maiyao1988/ExAndroidNativeEmu/files/8096422/libmynative-lib.zip)

**Comment by iamdrq** [comment]
My ENV:
```
OS: Windows 10 Pro 21H1 19043.1526
Python: 3.9 64bit
```
But it still fail that I try install python3.7 , 3.8 on Windows.
I also noticed pop `keystone.so error` when run test, I check the python source code `keystone` is useless on this project ? And I found `AndroidNativeEmu` install keystone-engine from requirememt.txt.

I guss maybe try this is a bad idea on Windows. so I try this on Linux:
```
ubuntu desktop vm (ubuntu 21.10 python 3.9.7) fail❌
ubuntu server (ubuntu 20.04 python 3.8.5) ok✅
```
Actually I purpose so is `yunceng.so`, It include `pthread_create` and `socket`(I notice `socket _connect return -1` in this project's source code), I am a freshman about this `ExAndroidNativeEmu`, Can it work ?
[libyunceng.so.zip](https://github.com/maiyao1988/ExAndroidNativeEmu/files/8101695/libyunceng.so.zip)
[yunceng.py.zip](https://github.com/maiyao1988/ExAndroidNativeEmu/files/8101721/yunceng.py.zip)
